libbeat/autodiscover/providers/kubernetes/kubernetes.go

```diff
@@ -217,7 +217,10 @@ func (p *Provider) emitEvents(pod *kubernetes.Pod, flag string, containers []kub
 
 		// This must be an id that doesn't depend on the state of the container
 		// so it works also on `stop` if containers have been already deleted.
-		eventID := fmt.Sprintf("%s.%s", pod.GetObjectMeta().GetUID(), c.Name)
+		eventID := fmt.Sprintf("%s", pod.GetObjectMeta().GetUID())
```

Comment:
Would this still work if we want to monitor more than one container per pod? For example if a pod has more than one container, each one exposing a port, and each one with a dedicated annotation. Something like this:
```
apiVersion: v1
kind: Pod
metadata:
  name: two-containers-prometheus
  annotations:
    co.elastic.metrics.prometheus-container/module: prometheus
    co.elastic.metrics.prometheus-container/hosts: ${data.host}:8080
    co.elastic.metrics.prometheus-container/module: redis
    co.elastic.metrics.prometheus-container/hosts: ${data.host}:6379
spec:
  restartPolicy: Never
  containers:
  - name: prometheus-container
    image: prom/prometheus
    ports:
      - containerPort: 8080
  - name: redis-container
    image: redis
    ports:
      - containerPort: 6379
```

Reply:
we have many use cases that use this feature. also, it would be necessary what container is logging the metric. We run filebeat and metricbeat on the same daemonset. since most of the metrics are same, we would need to be able to know what container the app is reporting from.